.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+        exclude: docs/
+    -   id: check-yaml
+    -   id: check-added-large-files
+-   repo: https://gitlab.com/pycqa/flake8.git
+    rev: 3.9.2
```

Comment:
Looks like the newest version is 4.0.1. Any reason to use older version?

Looking at [changelog](https://flake8.pycqa.org/en/latest/release-notes/4.0.0.html#backwards-incompatible-changes), I think we are safe to upgrade to 4.0.1. They removed VCS hooks integration (which is fine, that's what pre-commit is for), `setuptools` command (which is fine, as long as package can be installed) and support for user-level configuration files (`~/.flake8` and `~/.config/flake8` - which should force us to store configuration in repo and make it consistent for all contributors).